demo/demo/tests/test_apple_webhook.py

```diff
@@ -0,0 +1,264 @@
+    )
+    with mock.patch(RECEIPT_FETCH_FUNCTION) as mock_receipt:
+        mock_receipt.return_value = receipt_data
+        response = user_client.post(APPLE_API_WEBHOOK, make_receipt_query(), content_type='application/json')
+        assert response.status_code == 200
+
+    # Provide a notification with a different product id.
+    notification_data = make_notification_data(
+        apple_bundle_id,
+        apple_product_id + 'x',
+        original_transaction_id=transaction_id
+    )
+    with mock.patch(NOTIFICATION_PARSER) as mock_parser:
+        mock_parser.return_value = notification_data
+        with pytest.raises(ProductIdChangedError):
+            user_client.post(APPLE_API_WEBHOOK, make_notification_query(), content_type='application/json')
+
+
+def test__app_store_notifications__renew__subscription_extended(user_client,
+                                                                apple_bundle_id,
+                                                                apple_product_id,
+                                                                user,
+                                                                apple_in_app):
+    transaction_id = 'special-transaction-id'
+    # Create the subscription via API.
+    receipt_data = make_receipt_data(
+        apple_product_id,
+        apple_bundle_id,
+        transaction_id=transaction_id,
+        original_transaction_id=transaction_id
+    )
+    with mock.patch(RECEIPT_FETCH_FUNCTION) as mock_receipt:
+        mock_receipt.return_value = receipt_data
+        response = user_client.post(APPLE_API_WEBHOOK, make_receipt_query(), content_type='application/json')
+        assert response.status_code == 200, response.content
+
+    # Receive a notification about subscription extension.
+    notification_data = make_notification_data(
+        apple_bundle_id,
+        apple_product_id,
+        original_transaction_id=transaction_id
+    )
+    with mock.patch(NOTIFICATION_PARSER) as mock_parser:
+        mock_parser.return_value = notification_data
+        response = user_client.post(APPLE_API_WEBHOOK, make_notification_query(), content_type='application/json')
+        assert response.status_code == 200, response.content
+
+    all_subscriptions = list(SubscriptionPayment.objects.all())
+    assert len(all_subscriptions) == 2
+    payment = list(SubscriptionPayment.objects.all())[1]
+    transaction_info = notification_data.transaction_info
+    assert payment.user == user
+    assert payment.plan.metadata[apple_in_app.codename] == transaction_info.product_id
+    assert payment.status == SubscriptionPayment.Status.COMPLETED
+    assert payment.provider_codename == apple_in_app.codename
+    assert payment.provider_transaction_id == transaction_info.transaction_id
+    assert payment.subscription_start == transaction_info.purchase_date
+    assert payment.subscription_end == transaction_info.expires_date
+
+
+def test__app_store_notification__invalid_signature(client):
```

Comment:
Maybe use `pytest.mark.skip` to show explicitly that we skip this for now? And in all such cases too